vico/compose/src/main/java/com/patrykandpatrick/vico/compose/cartesian/CartesianChartHost.kt

```diff
@@ -174,63 +181,89 @@ internal fun CartesianChartHostImpl(
+                  null
                 }
-              } else {
-                null
-              }
-            },
-        )
-  ) {
-    val canvas = drawContext.canvas.nativeCanvas
-    if (canvas.width == 0 || canvas.height == 0) return@Canvas
-    canvasBounds.set(left = 0, top = 0, right = size.width, bottom = size.height)
-
-    layerDimensions.clear()
-    chart.prepare(measuringContext, layerDimensions)
-
-    if (chart.layerBounds.isEmpty) return@Canvas
-
-    zoomState.update(measuringContext, layerDimensions, chart.layerBounds)
-    scrollState.update(measuringContext, chart.layerBounds, layerDimensions)
-
-    if (model.id != previousModelID) {
-      coroutineScope.launch { scrollState.autoScroll(model, previousModel) }
-      previousModelID = model.id
+              },
+          )
+    ) {
+      val canvas = drawContext.canvas.nativeCanvas
+      if (canvas.width == 0 || canvas.height == 0) return@Canvas
+      canvasBounds.set(left = 0, top = 0, right = size.width, bottom = size.height)
+
+      layerDimensions.clear()
+      chart.prepare(measuringContext, layerDimensions)
+      targets = chart.allTargets
+
+      if (chart.layerBounds.isEmpty) return@Canvas
+
+      zoomState.update(measuringContext, layerDimensions, chart.layerBounds)
+      scrollState.update(measuringContext, chart.layerBounds, layerDimensions)
+
+      if (model.id != previousModelID) {
+        coroutineScope.launch { scrollState.autoScroll(model, previousModel) }
+        previousModelID = model.id
+      }
+
+      val context =
+        CartesianDrawingContext(
+            measuringContext,
+            canvas,
+            layerDimensions,
+            chart.layerBounds,
+            scrollState.value,
+            zoomState.value,
+          )
+          .also { drawingContext = it }
+
+      chart.draw(context)
+      measuringContext.reset()
     }
 
-    val drawingContext =
-      CartesianDrawingContext(
-        measuringContext,
-        canvas,
-        layerDimensions,
-        chart.layerBounds,
-        scrollState.value,
-        zoomState.value,
+    val context = drawingContext
+    if (isTouchExplorationEnabled() && context != null) {
+      AccessibilityHighlighter(
+        targets = targets,
+        context = context,
+        contentDescriptionProvider = chart.contentDescriptionProvider,
       )
-
-    chart.draw(drawingContext)
-    measuringContext.reset()
+    }
   }
 }
 
 @Composable
 private fun CartesianChartHostBox(modifier: Modifier, content: @Composable BoxScope.() -> Unit) {
   Box(modifier = modifier.height(CHART_HEIGHT.dp).fillMaxWidth(), content = content)
 }
+
+private val CartesianChart.allTargets: List<CartesianMarker.Target>
+  get() = layers.flatMap { it.markerTargets.values }.flatten()
+
+@Composable
+private fun isTouchExplorationEnabled(): Boolean {
+  val context = LocalContext.current
+  val accessibilityManager =
+    context.getSystemService(Context.ACCESSIBILITY_SERVICE) as? AccessibilityManager
```

Comment:
This should be wrapped with a `remember` block.
```suggestion
  val accessibilityManager =
    remember { context.getSystemService(Context.ACCESSIBILITY_SERVICE) as? AccessibilityManager }
```